pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -560,13 +622,13 @@ func (p *PriorityQueue) Update(oldPod, newPod *v1.Pod) error {
 				if err := p.podBackoffQ.Add(pInfo); err != nil {
 					return err
 				}
-				p.unschedulablePods.delete(usPodInfo.Pod)
+				p.unschedulablePods.delete(usPodInfo)
 				klog.V(5).InfoS("Pod moved to an internal scheduling queue", "pod", klog.KObj(pInfo.Pod), "event", PodUpdate, "queue", backoffQName)
 			} else {
-				if err := p.activeQ.Add(pInfo); err != nil {
+				if added, err := p.addToActiveQ(pInfo); !added {
```

Comment:
This should be `err != nil`

Reply:
nope, `!added` indicates either pod is gated, or error in adding pod to activeQ, so we should return in these cases.

`err != nil` missed the case that pod is gated.

Reply:
Make sense. Then if pod is gated, we should always return false in `isPodBackingoff`, right?  Or when backoff time arrives, we'll move pod to backoffQ.

Reply:
Another approach is similar to activeQ, we'll wrap the function of `p.podBackoffQ.Add()`, when it's pod gated, we'll move to unschedulablePods.

Reply:
What about this? @Huang-Wei

Reply:
Per offline discussion with @kerthcet, this is a very good point. And with current logic, it's problematic that a pod may be put to backoffQ in the following scenarios:

- pod's scheduling gates are removed within `podInitialBackoffSeconds` - in this case, isPodBackoff returns true and hence when a pod is updated, the pod is put into backoffQ
- when `movePodsToActiveOrBackoffQueue` is called (e.g., flushUnschedulablePodsLeftover or Node added), isPodBackingoff is also called and may returns true and move a gated pod to backoffQ.

I will append a fix in the next PR.